src/Storages/AlterCommands.cpp

```diff
@@ -1405,9 +1408,7 @@ void AlterCommands::validate(const StoragePtr & table, ContextPtr context) const
         }
         else if (command.type == AlterCommand::DROP_COLUMN)
         {
-            if (all_columns.has(command.column_name) ||
-                all_columns.hasNested(command.column_name) ||
-                (command.clear && column_name == LightweightDeleteDescription::FILTER_COLUMN.name))
+            if (all_columns.has(command.column_name) || all_columns.hasNested(command.column_name))
```

Comment:
Was the condition `command.clear && column_name == LightweightDeleteDescription::FILTER_COLUMN.name)` superfluous?

Reply:
Yes, it was superfluous. It was never satisfied because the query always fails earlier and it's the expected behaviour, I think.